server/auth_test.go

```diff
@@ -15,3 +15,60 @@ limitations under the License.
 */
 
 package server
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+var testServer = &Server{
+	Address: ":8081",
+	LDAP:    "",
+	Proxy:   "http://localhost:8080",
+}
+
+func TestServerLoginUnauthorized(t *testing.T) {
+	body, _ := json.Marshal(map[string]interface{}{"login": "tester", "password": "1234"})
+	req, err := http.NewRequest("POST", APIPrefix+"login", bytes.NewBuffer(body))
+	if err != nil {
+		t.Errorf("Unexpected error %s", err)
+	}
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(testServer.Login)
+	handler.ServeHTTP(rr, req)
+
+	// Check content-type header
+	if contentType := rr.Header().Get("Content-Type"); contentType != "application/json; charset=utf-8" {
+		t.Errorf("Expected Content-Type header value: %v but got %v",
+			"application/json; charset=utf-8", contentType)
+	}
+
+	// Check the status code
+	if status := rr.Code; status != http.StatusUnauthorized {
+		t.Errorf("Expected status code: %v but got %v",
+			http.StatusUnauthorized, status)
+	}
+
+}
+
+func TestJwtAuthNoProxy(t *testing.T) {
+	body, _ := json.Marshal(map[string]interface{}{"login": "tester", "password": "1234"})
+	req, err := http.NewRequest("POST", APIPrefix+"login", bytes.NewBuffer(body))
+	if err != nil {
+		t.Errorf("Unexpected error %s", err)
+	}
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(testServer.HandleHTTP)
+	newHandler := testServer.JWTAuthentication(handler)
+	newHandler.ServeHTTP(rr, req)
+
+	// Check the status code
+	if status := rr.Code; status != http.StatusServiceUnavailable {
+		t.Errorf("Expected status code: %v but got %v",
+			http.StatusServiceUnavailable, status)
+	}
+
+}
```

Comment:
I am not sure this test makes some sense.It relies on fact that nothing is running at `localhost:8080`.

It's increasing code coverage at least.... :)